src/main/java/datawave/microservice/config/security/AllowedCallersFilter.java

```diff
@@ -0,0 +1,65 @@
+package datawave.microservice.config.security;
+
+import datawave.microservice.authorization.config.DatawaveSecurityProperties;
+import datawave.security.authorization.SubjectIssuerDNPair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.lang.Nullable;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+public class AllowedCallersFilter extends OncePerRequestFilter {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final DatawaveSecurityProperties securityProperties;
+    
+    public AllowedCallersFilter(DatawaveSecurityProperties securityProperties) {
+        this.securityProperties = securityProperties;
+    }
+    
+    @Override
+    protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain)
+                    throws ServletException, IOException {
+        if (securityProperties.isEnforceAllowedCallers()) {
+            // Extract the client certificate, and if one is provided, validate that the caller is allowed to talk to us.
+            X509Certificate cert = extractClientCertificate(httpServletRequest);
+            if (cert != null) {
```

Comment:
Still trying to decide if we should be returning a 403, if enforcement is enabled and we hit the case where (cert == null). On the surface, it seems like we should raise a 403 but I'm on the fence, given that the code has always worked this way...and I'm unsure whether there are valid request contexts where those two conditions can be true